File: QE-Selenium python/Activity8.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.firefox.service import Service
from webdriver_manager.firefox import GeckoDriverManager
import time

driver = webdriver.Firefox(service=Service(GeckoDriverManager().install()))
driver.maximize_window()

driver.get("https://training-support.net/webelements/mouse-events")


print("Page Title:", driver.title)

actions = ActionChains(driver)

cargo_lock = driver.find_element(By.ID, "cargo-lock")
cargo_toml = driver.find_element(By.ID, "cargo-toml")
src = driver.find_element(By.ID, "src")
target = driver.find_element(By.ID, "target")
open_option = driver.find_element(By.ID, "open")


actions.click(cargo_lock).move_to_element(cargo_toml).click().perform()
time.sleep(1)
print("Output 1:", driver.find_element(By.ID, "result").text)


actions.double_click(src).perform()
time.sleep(1)


actions.context_click(target).click(open_option).perform()
time.sleep(1)
print("Output 2:", driver.find_element(By.ID, "result").text)

driver.quit()

File: QE-Selenum java/demo/src/main/java/demos/Activity8.java
package demos;

public class Activity8 {
    from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.chrome.service import Service
import time

# Setup WebDriver
driver = webdriver.Chrome()
driver.maximize_window()

# Open URL
driver.get("https://www.hithips.com/training-support.net/webelements/mouse-events")

# Print page title
print("Page Title:", driver.title)

actions = ActionChains(driver)

# -------------------------------
# LEFT CLICK SEQUENCE
# -------------------------------

cargo_lock = driver.find_element(By.ID, "cargo-lock")
cargo_toml = driver.find_element(By.ID, "cargo-toml")

actions.click(cargo_lock).pause(1).move_to_element(cargo_toml).click().perform()

# Get confirmation text
confirmation1 = driver.find_element(By.ID, "result").text
print("Confirmation after cargo sequence:", confirmation1)

time.sleep(2)

# -------------------------------
# DOUBLE CLICK + RIGHT CLICK
# -------------------------------

are_button = driver.find_element(By.ID, "are")
target_bullseye = driver.find_element(By.ID, "target")

actions.double_click(are_button).perform()
time.sleep(1)

actions.context_click(target_bullseye).perform()
time.sleep(1)

# Click "Open" from context menu
open_option = driver.find_element(By.XPATH, "//li[text()='Open']")
open_option.click()

# Get confirmation text
confirmation2 = driver.find_element(By.ID, "result").text
print("Confirmation after double/right click sequence:", confirmation2)

# Close browser
time.sleep(2)
driver.quit()


}
